src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,169 @@
+            System.out.println(value.toString());
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> node = getNodeByIndex(index);
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNodeByIndex(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = getNodeByIndex(index);
+        T value = node.item;
+        unlink(node);
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = firstNode;
+        for (int i = 0; i < size; i++) {
+            if (isItemEquals(node.item, object)) {
+                unlink(node);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(MyLinkedList.Node<T> prev, T element, MyLinkedList.Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void unlink(Node<T> node) {
+        if (size == 1) {
+            firstNode = null;
+        } else {
+            if (node.equals(firstNode)) {
+                node.next.prev = null;
+                firstNode = node.next;
+            } else if (node.equals(lastNode)) {
+                node.prev.next = null;
+                lastNode = node.prev;
+            } else {
+                node.prev.next = node.next;
+                node.next.prev = node.prev;
+            }
+        }
+        size--;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == 0) {
+            return firstNode;
+        }
+        if (index == size - 1) {
+            return lastNode;
+        }
+        return (index <= size >> 1) ? getNodeFromHead(index) : getNodeFromTail(index);
+    }
+
+    private Node<T> getNodeFromHead(int index) {
+        Node<T> node = firstNode;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return node;
+    }
+
+    private Node<T> getNodeFromTail(int index) {
+        Node<T> node = lastNode;
+        for (int i = index; i < size - 1; i++) {
+            node = node.prev;
+        }
+        return node;
+    }
+
+    private boolean isItemEquals(T firstItem, T secondItem) {
+        return Objects.equals(firstItem, secondItem);
+    }
```

Comment:
Don’t use class Objects (according to checkstyle) 
I think you can wtie your implementation:)